Sources/Build/Buildable.swift

```diff
@@ -59,16 +59,24 @@ extension Module: Buildable {
     }
 }
 
-extension SwiftModule {
-    func pkgConfigArgs() throws -> [String] {
-        return try recursiveDependencies.flatMap { module -> [String] in
+extension ModuleProtocol {
+
+    func pkgConfigArgsCombined() throws -> [String] {
```

Comment:
I think we should start with a whitelist of arguments we know are ok (-I, -L, -l) and disallow packages using anything else. Then the diagnostics are at least explicit, and we can figure out the "right thing" to do about new flags as they appear.

We could also whitelist -I, -L, -l to go to swiftc directly, and send everything else through `-Xlinker`.

The downside of simply sending everything through `-Xlinker` is that what we usually really want to do is pass the arguments to the GCC-compatible driver _when used as a linker_, which is not currently what `swiftc` does with `-Xlinker`, and it also doesn't have any other option which does that.